src/compat/__tests__/should_wait_for_data_before_loaded.test.ts

```diff
@@ -67,7 +67,55 @@ describe("compat - shouldWaitForDataBeforeLoaded", () => {
     });
     const shouldWaitForDataBeforeLoaded =
       require("../should_wait_for_data_before_loaded");
-    expect(shouldWaitForDataBeforeLoaded.default(true)).toBe(false);
+    expect(shouldWaitForDataBeforeLoaded.default(true, false)).toBe(false);
+  });
+
+  it("should return true if we are on the Safari browser, we should play inline and in directfile mode", () => {
+    jest.mock("../browser_detection", () => {
+      return {
+        __esModule: true,
+        isSafariMobile: true,
+      };
+    });
+    const shouldWaitForDataBeforeLoaded =
+      require("../should_wait_for_data_before_loaded");
+    expect(shouldWaitForDataBeforeLoaded.default(true, true)).toBe(true);
+  });
+
+  it("should return true if we are on the Safari browser, we should play inline but not in directfile mode", () => {
+    jest.mock("../browser_detection", () => {
+      return {
+        __esModule: true,
+        isSafariMobile: true,
+      };
+    });
+    const shouldWaitForDataBeforeLoaded =
+      require("../should_wait_for_data_before_loaded");
+    expect(shouldWaitForDataBeforeLoaded.default(false, true)).toBe(true);
+  });
+
+  it("should return false if we are on the Safari browser, we should not play inline and in directfile mode", () => {
```

Comment:
This test already existed before. Just change the description of the other test

Reply:
Also `we should not play inline and in directfile mode` -> `we should not play inline but in directfile mode`?
Here it sounds like we should not play in directfile mode.

Reply:
yes, my bad